### ISSUE TITLE
Returning Sorted Table to Default

### ISSUE THREAD

**Issue by vinceclicks** [primefaces/primeng#12553]
### Describe the feature you would like to see added

Sortable table columns are currently boolean, a two click system: ascending and descending.

The table sort headers should have 3 clicks, or an option for 3 clicks, to turn off the sort and return the default data.

This has been requested for years and seems to have been closed before [(3217)](https://github.com/primefaces/primeng/issues/3217) because of switches from datatable to table or other weird things. I'd like to reintroduce the request.

### Is your feature request related to a problem?

The table sort doesn't have an option to return to default.

### Describe the solution you'd like

Table column sorting should be clickable a third time, returning the data to default.

### Describe alternatives you have considered

My team has pointed to the material library as a possibility.

**Comment by smootyDev** [comment]
+1

**Comment by glqennova** [comment]
Is actually uncommon find tables that doesn't have the sort reset at 3rd click, i wonder why they dont add this new feature, it is actually easy to set

**Comment by hugovlc** [comment]
I agree. Makes no sense not having a state with no value. We might want to sort by a different field and forget about the previous one. I don´t wanna code special cases in which I have to reset the state of the table. Please consider doing this. Is common sense.

**Comment by fateeand** [comment]
Really frustrating experience of not having such a feature

**Comment by Jannis033** [comment]
+1
Maybe it can be implemented as an optional feature that you have to enable it for a table first, so nothing gets broken.

**Comment by GusBeare** [comment]
Any news on this?  Just discovered it. Why anyone would build and release a sort feature without any option to "reset" is hard to understand.

**Comment by Jannis033** [comment]
I got it working by using this workaround: [https://stackblitz.com/edit/primeng-tablesort-demo-hz7lz8?file=src%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/primeng-tablesort-demo-hz7lz8?file=src%2Fapp%2Fapp.component.ts)
However, to handle data changes dynamically you will need a bit more code and also, large amounts of data could cause performance issues. This may the the reason why PrimeNg did not implement it, idk.
I still used this approach but renamed initialValue to sortedValue and added this variable to the table instead of the original data. If the data gets changed, I just check for the value of isSorted and copy the new data (sorted or not, depending on isSorted) to sortedValue (deep copy, not "linking" to avoid changing the initial data!) . 
Furthermore, you may have to add a check to customSort() that checks whether a new column has been sorted, because otherwise it will just cycle through the three states, regardless of whether the same or another column was clicked.

**Comment by mh-zwave** [comment]
+1

**Comment by BergkampHUN** [comment]
Thanks @Jannis033, sadly this solution works only when there aren't any filter in use, then the initialValue overwrites it.

**Comment by fkrajcar-cirtuo** [comment]
+1
